src/gt4py/next/program_processors/runners/dace_fieldview/workflow.py

```diff
@@ -45,9 +49,18 @@ def generate_sdfg(
         ir: itir.Program,
         offset_provider: common.OffsetProvider,
         column_axis: Optional[common.Dimension],
+        auto_opt: bool,
+        on_gpu: bool,
     ) -> dace.SDFG:
         ir = itir_transforms.apply_fieldview_transforms(ir, offset_provider=offset_provider)
-        return gtir_sdfg.build_sdfg_from_gtir(ir=ir, offset_provider=offset_provider)
+        sdfg = gtir_sdfg.build_sdfg_from_gtir(ir, offset_provider=offset_provider)
+
+        if auto_opt:
+            return gtx_transformations.gt_auto_optimize(sdfg, gpu=on_gpu)
```

Comment:
A single return statement does look better and is less confusing.

```suggestion
            gtx_transformations.gt_auto_optimize(sdfg, gpu=on_gpu)
```

Reply:
Right, I forgot that it operates in place.